File: src/main/java/edu/spbstu/taxi/controller/OperatorController.java
package edu.spbstu.taxi.controller;

import edu.spbstu.taxi.Exceptions.HaveNotOrderEx;
import edu.spbstu.taxi.Exceptions.HaveNotUserEx;
import edu.spbstu.taxi.entity.Driver;
import edu.spbstu.taxi.entity.Operator;
import edu.spbstu.taxi.entity.Order;
import edu.spbstu.taxi.service.ServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.List;

@RestController
public class OperatorController {
    @Autowired
    ServiceImpl service;

    @RequestMapping("rest/operator/{login}/authenticate")
    public String authenticate(@PathVariable String login, @RequestParam("passwd") String passwd) {
        if (service.authenticate(login, passwd) != 0)
            return "false";
        return "true";
    }

    @RequestMapping(value = "rest/operator/{login}", method = RequestMethod.POST)
    public void newOperator(@PathVariable String login, @RequestBody Operator op) {
       service.addNewOperator(op);
    }

    /*@RequestMapping(value = "rest/operator/{login}", method = RequestMethod.GET)
    public Operator getOperator(@PathVariable String login) {
        service.findOperatorByLogin(login).orElseThrow(() -> new.. .//???
    }*/


    @GetMapping("rest/operator/{login}/drivers")
    public @ResponseBody
    List<Driver> getAvailableDrivers(@PathVariable String login){
        try {
            return service.getAvailableDrivers(login);
        } catch (HaveNotUserEx ex){
            return null;
        }
    }

    @GetMapping("rest/operator/{login}/orders")
    public @ResponseBody
    List<Order> getNewOrders(@PathVariable String login){
        try {
            return service.getNewOrders(login);
        } catch (HaveNotOrderEx ex){
            return new ArrayList<Order>();
        } catch (HaveNotUserEx ex){
            return new ArrayList<Order>();
        }
    }

    @GetMapping(value = "rest/operator/{login}/appoint")
    public String appointOrder(@PathVariable String login, @RequestParam("OrID") int OrID,@RequestParam("DrID") int DrID) {
        try {
            service.appointOrdertoDriver(OrID, DrID, login);
            return "true";
        }catch (HaveNotOrderEx ex){
            return "false";
        }catch( HaveNotUserEx ex){
            return "false";
        }
    }

}

File: src/main/webapp/resources/js/operatorController.js
function OperatorService($resource) {
    return $resource('rest/operator/:login/:reqType?OrID=:OrID&DrID=:DrID',
        {reqType: '@reqType',  login: '@login', OrID:'@OrID', DrID:'@DrID' });
}

function OperatorCtrl($scope, $http, OperatorService, InfoShareService){
    $scope.login = InfoShareService.getUser();
    OperatorService.query({login:$scope.login, reqType: "orders"}, function (value){$scope.orders = value;});
    $scope.drivers = OperatorService.query({login:$scope.login, reqType: "drivers"});

    $scope.select = function(value){
        alert("Order selected!");
        $scope.OrID= value;
    };
    $scope.selectDr = function(value){
        alert("Driver selected!");
        $scope.DrID= value;
    };
    $scope.update = function(){
        alert("update");
        OperatorService.query({login:$scope.login, reqType: "orders"},
            function (value){$scope.orders = value;});
        $scope.drivers = OperatorService.query({login:$scope.login, reqType: "drivers"});
    };
    $scope.appoint = function () {
        alert("appoint");
        if (!$scope.OrID){
            alert("choose order!");
        } else if (!$scope.DrID){
            alert("choose driver!");
        } else {

            $scope.OK = OperatorService.query({login:$scope.login,
                reqType: "appoint", OrID:$scope.OrID, DrID:$scope.DrID});
            if($scope.OK) {
                $scope.update();
            }
        }
    };
}


app
    .factory('OperatorService', OperatorService)
    .controller('OperatorCtrl', OperatorCtrl);